app/src/test/kotlin/fr/free/nrw/commons/contributions/ContributionsPresenterTest.kt

```diff
@@ -27,84 +37,58 @@ class ContributionsPresenterTest {
 
     lateinit var loader: Loader<Cursor>
 
+    lateinit var liveData: LiveData<List<Contribution>>
+
+    @Rule @JvmField var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    lateinit var scheduler : Scheduler
+
     /**
      * initial setup
      */
     @Before
     @Throws(Exception::class)
     fun setUp() {
         MockitoAnnotations.initMocks(this)
+        scheduler=TestScheduler()
         cursor = Mockito.mock(Cursor::class.java)
         contribution = Mockito.mock(Contribution::class.java)
-        contributionsPresenter = ContributionsPresenter(repository)
+        contributionsPresenter = ContributionsPresenter(repository,scheduler,scheduler)
         loader = Mockito.mock(CursorLoader::class.java)
         contributionsPresenter?.onAttachView(view)
+        liveData=MutableLiveData()
     }
 
-
     /**
-     * Test presenter actions onGetContributionFromCursor
+     * Test fetch contributions
      */
     @Test
-    fun testGetContributionFromCursor() {
-        contributionsPresenter?.getContributionsFromCursor(cursor)
-        verify(repository)?.getContributionFromCursor(cursor)
+    fun testFetchContributions(){
+        Mockito.`when`(repository?.getString(ArgumentMatchers.anyString())).thenReturn("10")
+        Mockito.`when`(repository?.fetchContributions()).thenReturn(liveData)
+        contributionsPresenter?.fetchContributions()
```

Comment:
Why is `contributionsPresenter` & `repository` nullable? Fix that if you could.

Reply:
This was the existing implementation, but fair point, I had already touched this, fixed that, thanks for pointing it out